api-audit/src/main/java/com/capitalone/dashboard/evaluator/RegressionTestResultEvaluator.java

```diff
@@ -4,77 +4,342 @@
+import com.capitalone.dashboard.model.TestSuite;
+import com.capitalone.dashboard.model.TestCase;
+import com.capitalone.dashboard.model.Feature;
+import com.capitalone.dashboard.model.StoryIndicator;
+import com.capitalone.dashboard.repository.FeatureRepository;
 import com.capitalone.dashboard.repository.TestResultRepository;
 import com.capitalone.dashboard.response.TestResultsAuditResponse;
 import com.capitalone.dashboard.status.TestResultAuditStatus;
 import org.apache.commons.collections.CollectionUtils;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 import java.util.ArrayList;
 import java.util.Collection;
-import java.util.Comparator;
 import java.util.List;
 import java.util.Map;
-import java.util.stream.Collectors;
+import java.util.Set;
+import java.util.HashMap;
+import java.util.Date;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+
 
 @Component
 public class RegressionTestResultEvaluator extends Evaluator<TestResultsAuditResponse> {
 
     private final TestResultRepository testResultRepository;
-
+    private final FeatureRepository featureRepository;
+    private static final Logger LOGGER = LoggerFactory.getLogger(RegressionTestResultEvaluator.class);
 
     @Autowired
-    public RegressionTestResultEvaluator(TestResultRepository testResultRepository) {
+    public RegressionTestResultEvaluator(TestResultRepository testResultRepository, FeatureRepository featureRepository) {
         this.testResultRepository = testResultRepository;
+        this.featureRepository = featureRepository;
     }
 
     @Override
     public Collection<TestResultsAuditResponse> evaluate(Dashboard dashboard, long beginDate, long endDate, Map<?, ?> dummy) throws AuditException {
+
         List<CollectorItem> testItems = getCollectorItems(dashboard, "codeanalysis", CollectorType.Test);
-        Collection<TestResultsAuditResponse> responses = new ArrayList<>();
+        Collection<TestResultsAuditResponse> testResultsAuditResponse = new ArrayList<>();
+
         if (CollectionUtils.isEmpty(testItems)) {
             throw new AuditException("No tests configured", AuditException.NO_COLLECTOR_ITEM_CONFIGURED);
         }
 
-        return testItems.stream().map(item -> evaluate(item, beginDate, endDate, null)).collect(Collectors.toList());
+        testItems.forEach(testItem -> {
+            testResultsAuditResponse.add(getRegressionTestResultAudit(dashboard, testItem, beginDate, endDate));
+
+        });
+
+        return testResultsAuditResponse;
     }
 
     @Override
     public TestResultsAuditResponse evaluate(CollectorItem collectorItem, long beginDate, long endDate, Map<?, ?> dummy) {
-        return getRegressionTestResultAudit(collectorItem, beginDate, endDate);
+        return null;
```

Comment:
Why is this returning null?

Reply:
because thats an abstract method in Evaluator class and we don't need any implementation for it in RegressionTestResultsEvaluator